cnf-certification-test/networking/declaredandlistening_test.go

```diff
@@ -0,0 +1,328 @@
+// Copyright (C) 2020-2021 Red Hat, Inc.
+//
+// This program is free software; you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation; either version 2 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License along
+// with this program; if not, write to the Free Software Foundation, Inc.,
+// 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+
+package declaredandlistening
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/test-network-function/cnf-certification-test/cnf-certification-test/networking/declaredandlistening"
+	"github.com/test-network-function/cnf-certification-test/pkg/provider"
+	v1 "k8s.io/api/core/v1"
+)
+
+//nolint:dupl,funlen
+func TestParseVariables(t *testing.T) {
+	// expected inputs
+	testCases := []struct {
+		// inputRes is string that include the result after we run the command ""oc get pod %s -n %s -o json  | jq -r '.spec.containers[%d].ports'""
+		inputRes string
+		// now is empty but maybe in the future has be not empty.
+		listeningPorts map[declaredandlistening.Key]*provider.Container
+		container      *provider.Container
+		// expected outputs here
+		expectedlisteningPorts map[declaredandlistening.Key]*provider.Container
+		expectedRes            string
+		expectContainer        *provider.Container
+	}{
+		{
+			inputRes:       "tcp LISTEN 0      128    0.0.0.0:8080 0.0.0.0:*\n",
+			listeningPorts: map[declaredandlistening.Key]*provider.Container{},
```

Comment:
You don't need this listeningPorts field in the tc struct, as you can create it for each iteration. See comment below.